src/main/java/com/gotocompany/firehose/sink/grpc/GrpcSinkFactory.java

```diff
@@ -21,24 +29,56 @@
  * using the configurations supplied and invoke {@see #create(Map < String, String > configuration, StatsDClient client)}
  * to obtain the GrpcSink sink implementation.
  */
+@Slf4j
 public class GrpcSinkFactory {
 
+    private static final String CERT_DIRECTORY = "/usr/local/share/ca-certificates";
+
+    private static File createCertFileFromBase64(String base64Cert) throws IOException {
+        byte[] decodedBytes = Base64.getDecoder().decode(base64Cert);
+        File certDirectory = new File(CERT_DIRECTORY);
+        if (!certDirectory.exists()) {
+            certDirectory.mkdirs();
+        }
+        File certFile = Files.createTempFile(certDirectory.toPath(), "client-cert", ".crt").toFile();
+        log.info("Created certificate file at {}", certFile.getAbsolutePath());
+        try (FileOutputStream fos = new FileOutputStream(certFile)) {
+            fos.write(decodedBytes);
+        }
+        return certFile;
+    }
+
+    private static SslContext buildClientSslContext(String base64Cert) {
+        try {
+            File certFile = createCertFileFromBase64(base64Cert);
+            SslContextBuilder sslContextBuilder = SslContextBuilder.forClient().trustManager(certFile);
+            return GrpcSslContexts
+                    .configure(sslContextBuilder)
+                    .build();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
throw new RuntimeException(e);, RuntimeException("failed to create ssl ......", e)